### PR TITLE
AudioReceiveHandler#includeUserInCombinedAudio

### PR DESCRIPTION
**Pull request by DV8FromTheWorld**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: #967

## Description
Adds support for filtering what users are included in in the packet created for `handleCombinedAudio(CombinedAudio)`

Example: Only include audio from users that are not bots.
```
public MyHandler implements AudioReceiveHandler {
  canReceiveCombined() {
    return true;
  }

  handleCombinedAudio(CombinedAudio audio) {
    //Handle audio
  }

  includeUserInCombinedAudio(User user) {
    return !user.isBot();
  }
}
```

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java [hunk 1]
```diff
@@ -86,4 +88,26 @@
...
     void handleUserAudio(UserAudio userAudio);
+
+    /**
+     * This method is a filter predicate used by JDA to determine whether or not to include a
+     * {@link net.dv8tion.jda.api.entities.User User}'s audio when creating a CombinedAudio packet.
+     * <p>
+     * This method is especially useful in creating whitelist / blacklist functionality for receiving audio.
+     * <p>
+     * A few possible examples:
+     *  <li>Have this method always return false for Users that are bots.</li>
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java]: Isn't this missing a `<ul>` somewhere?

File: src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java [hunk 2]
```diff
@@ -86,4 +88,26 @@
...
+     *  <li>Have this method return false for users who have been placed on a blacklist for abusing the bot's functionality.</li>
+     *  <li>Have this method only return true if the user is in a special whitelist of power users.</li>
+     *
+     * @param  user
+     *         The user whose audio was received
+     *
+     * @return If true, JDA will include the user's audio when merging audio sources when created packets
+     *         for {@link #handleCombinedAudio(CombinedAudio)}
+     */
+    default boolean includeUserInCombinedAudio(User user)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java]: This should have a `@Nonnull` on the user parameter

File: src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java (hunk 1 above)
Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/audio/AudioReceiveHandler.java]: Oh. Yeah, you're right.
I guess the javadoc renderer just understood the lack of `<ul>` when I previewed it.

**Comment by MinnDevelopment** [comment]
The name of this method seems pretty obnoxious to me, can't we come up with something more slim?

**Comment by DV8FromTheWorld** [comment]
I am open to suggestions as the current name is straight forward in its usage

**Comment by DV8FromTheWorld** [comment]
Is there a counter suggestion for method naming? If not, we should progress this PR

**Pull request merged**